controllers/argocd/sso_test.go

```diff
@@ -75,3 +80,106 @@ func TestReconcile_noTemplateInstance(t *testing.T) {
 
 	assert.NilError(t, r.reconcileSSO(a))
 }
+
+func TestReconcile_testKeycloakK8sInstance(t *testing.T) {
+	logf.SetLogger(ZapLogger(true))
+	a := makeTestArgoCDForKeycloak()
+
+	// Cluster does not have a template instance
+	templateAPIFound = false
+	r := makeReconciler(t, a)
+
+	assert.NilError(t, r.reconcileSSO(a))
+}
+
+func TestReconcile_testKeycloakInstanceResources(t *testing.T) {
+	logf.SetLogger(ZapLogger(true))
+	a := makeTestArgoCDForKeycloak()
+
+	// Cluster does not have a template instance
+	templateAPIFound = false
+	r := makeReconciler(t, a)
+
+	assert.NilError(t, r.reconcileSSO(a))
+
+	// Keycloak Deployment
+	deployment := &k8sappsv1.Deployment{}
+	err := r.Client.Get(context.TODO(), types.NamespacedName{Name: defaultKeycloakIdentifier, Namespace: a.Namespace}, deployment)
+	assert.NilError(t, err)
+
+	assert.Equal(t, deployment.Name, defaultKeycloakIdentifier)
+	assert.Equal(t, deployment.Namespace, a.Namespace)
+
+	testLabels := map[string]string{
+		"app": defaultKeycloakIdentifier,
+	}
+	assert.DeepEqual(t, deployment.Labels, testLabels)
+
+	testSelector := &v1.LabelSelector{
+		MatchLabels: map[string]string{
+			"app": defaultKeycloakIdentifier,
+		},
+	}
+	assert.DeepEqual(t, deployment.Spec.Selector, testSelector)
+
+	assert.DeepEqual(t, deployment.Spec.Template.ObjectMeta.Labels, testLabels)
+	assert.Equal(t, deployment.Spec.Template.Spec.Containers[0].Name,
+		defaultKeycloakIdentifier)
+	assert.Equal(t, deployment.Spec.Template.Spec.Containers[0].Image,
+		getKeycloakContainerImage(a))
+
+	testEnv := []corev1.EnvVar{
+		{Name: "KEYCLOAK_USER", Value: defaultKeycloakAdminUser},
+		{Name: "KEYCLOAK_PASSWORD", Value: defaultKeycloakAdminPassword},
+		{Name: "PROXY_ADDRESS_FORWARDING", Value: "true"},
+	}
+	assert.DeepEqual(t, deployment.Spec.Template.Spec.Containers[0].Env,
+		testEnv)
+
+	// Keycloak Service
+	svc := &corev1.Service{}
+	err = r.Client.Get(context.TODO(), types.NamespacedName{Name: defaultKeycloakIdentifier, Namespace: a.Namespace}, svc)
+	assert.NilError(t, err)
+
+	assert.Equal(t, svc.Name, defaultKeycloakIdentifier)
+	assert.Equal(t, svc.Namespace, a.Namespace)
+	assert.DeepEqual(t, svc.Labels, testLabels)
+
+	assert.DeepEqual(t, svc.Spec.Selector, testLabels)
+	assert.DeepEqual(t, svc.Spec.Type, corev1.ServiceType("LoadBalancer"))
+
+	// Keycloak Ingress
+	ing := &extv1beta1.Ingress{}
+	err = r.Client.Get(context.TODO(), types.NamespacedName{Name: defaultKeycloakIdentifier, Namespace: a.Namespace}, ing)
+	assert.NilError(t, err)
+
+	assert.Equal(t, ing.Name, defaultKeycloakIdentifier)
+	assert.Equal(t, ing.Namespace, a.Namespace)
+
+	testTLS := []extv1beta1.IngressTLS{
```

Comment:
We migrated from `extensions.k8s.io/v1beta1` to `networking.k8s.io/v1` for Ingress. This should be adapted for in all tests.

Reply:
made the required changes,